service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -92,6 +95,50 @@ void shouldReturnCaseManagementOrderWhenProvidedCaseDetails() {
             formatLocalDateToMediumStyle(5));
     }
 
+    @Test
+    void shouldFormatRespondentsIntoKeyWhenRespondentsArePresent() {
+        CaseDetails caseDetails = CaseDetails.builder()
```

Comment:
Nasty consequence of accepting `CaseDetails` everywhere is that it needs to be created in test, quite pointless. If correctly implemented method would just take list of respondents. It would not care about case details nor case data. It would not have to know how to retrieve relevant details from 'context' objects.